src/services/preProcess.ts

```diff
@@ -345,6 +345,42 @@ namespace ts {
                     break;
                 }
 
+                if (scanner.getToken() === SyntaxKind.TemplateHead) {
+                    const stack = [scanner.getToken()];
+                    loop: while (length(stack)) {
+                        const token = scanner.getToken();

```

Comment:
You're not advancing the scanner after your first call to `getToken()` so does that mean you always end up pushing 2 `TemplateHead`s?